src/collections/componentSet.ts

```diff
@@ -204,6 +206,37 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
     return result;
   }
 
+  /**
+   * Resolve components from an org connection.
+   *
+   * @param connection org connection
+   * @returns ComponentSet of source resolved components
+   */
+  public static async fromConnection(connection: Connection): Promise<ComponentSet>;
```

Comment:
I'm not sure here. I tried to define the method the same way `fromManifest` is defined.

Reply:
supporting `FromConnectionOptions` would be a more flexible, "future-proof" approach, but I cant think of any other options to pass here.  It would be easy to add filtering options here if that would be useful so maybe?

In terms of method overloading, I'd like to see a string arg that is the org username so that SDR would create the connection.  So e.g.,
`public static async fromConnection(connection: Connection | orgUsername: string): Promise<ComponentSet>`

Reply:
@shetzel should we create the connection based on the username in the `fromConnection` method or the `ConnectionResolver` class?

Reply:
@shetzel @mshanemc I've updated the `FromConnection` method to match it with the `fromManifest` method. It now has a username as string and `FromConnectionOptions` covers `usernameOrConnection: string | Connection;` and `apiVersion?: string;` 

`public static async fromConnection(input: string | FromConnectionOptions): Promise<ComponentSet>`